chia/cmds/coin_funcs.py

```diff
@@ -127,27 +128,37 @@ async def async_combine(args: Dict[str, Any], wallet_client: WalletRpcClient, fi
     final_min_coin_amount: uint64 = uint64(int(min_coin_amount * mojo_per_unit))
     final_excluded_amounts: List[uint64] = [uint64(int(amount * mojo_per_unit)) for amount in excluded_amounts]
     final_fee = uint64(int(fee * mojo_per_unit))
-    conf_coins, _, _ = await wallet_client.get_spendable_coins(
-        wallet_id=wallet_id,
-        max_coin_amount=final_max_dust_amount,
-        min_coin_amount=final_min_coin_amount,
-        excluded_amounts=final_excluded_amounts,
-    )
-    if len(target_coin_ids) > 0:
-        conf_coins = [cr for cr in conf_coins if cr.name in target_coin_ids]
-    if len(conf_coins) <= 1:
-        print("No coins to combine.")
-        return
-    if largest:
-        conf_coins.sort(key=lambda r: r.coin.amount, reverse=True)
+    final_target_coin_amount = uint64(int(target_coin_amount * mojo_per_unit))
+    if final_target_coin_amount != 0:  # if we have a set target, just use standard coin selection.
+        removals: List[Coin] = await wallet_client.select_coins(
+            amount=final_target_coin_amount + final_fee,
+            wallet_id=wallet_id,
+            max_coin_amount=final_max_dust_amount,
+            min_coin_amount=final_min_coin_amount,
+            excluded_amounts=final_excluded_amounts + [final_target_coin_amount],  # dont reuse coins of same amount.
```

Comment:
this comment is cryptic. I don't understand what it's trying to say

Reply:
if we have a coin of amount a but we want to make another coin of amount a, the default coin selection algorithm will just use that existing coin of amount a, instead of choosing a set of new coins to create amount a. So we tell the algorithm to filter out all coins of amount a so that the coins outputted from the algorithm can result in amount a but don't only contain a coin of amount a.